packages/transit-vehicle-overlay/src/VehicleGeometry/index.js

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { FeatureGroup } from "react-leaflet";
+import { leafletPathType } from "@opentripplanner/core-utils/lib/types";
+import { vehicleType } from "../types";
+import * as utils from "./utils";
+import { setColor } from "../utils";
+
+/**
+ * vehicle geometry presentational component that creates a map overlay for the line
+ * geometry showing the travel pattern of a vehicle
+ */
+function VehicleGeometry(props) {
+  const { trackedVehicle } = props;
+  const { pattern } = props;
+  const { lowlight, color } = props;
+  let { highlight } = props;
+  if (color) {
+    highlight = setColor(color, highlight);
+  }
+
+  let retVal = <FeatureGroup />;
```

Comment:
This could be moved to after the following if block and the FeatureGroup with the segments could be immediately returned.